src/mocker.js

```diff
@@ -1,14 +1,16 @@
 const { loadAndSetUserConfigurations, config } = require('./config');
 const { processVocabulary } = require('./vocabulary');
-const { generateCases, generateEvents } = require('./data');
+const { generateCases, generateEvents, generateDynamicCaseAttrs } = require('./data');
 const { saveToCSV, writeFile } = require('./output');
 const { generateSchemaSql, generateSqlInsert } = require('./sql_generator');
 const util = require('./util');
 const pluralize = require('pluralize');
 
 async function main() {
   loadAndSetUserConfigurations();
-  const vocabulary = processVocabulary();
+
+  const dynamicCaseAttrs = generateDynamicCaseAttrs(config.DYNAMIC_ATTRS, config.UNIQUE_VALUES_FOR_DYNAMIC_ATTRS);
+  const vocabulary = processVocabulary(dynamicCaseAttrs);
```

Comment:
I wouldn't pass the dynamic values from here. Instead, I'd delegate it to the `processVocabulary()` function to adjust its schema when dynamic attributes are being applied.

Reply:
When I call `generateDynamicCaseAttrs` from `vocabulary.js`, it introduced a cyclic dependency between `vocabulary.js` and `data.js`. To avoid this, I had to move the function definition from `data.js` to `vocabulary.js`. This didn't look ideal (since it is generating data), but I could't find a way around it (it was either this or create a new js file to define this method, which I felt is little excessive).